services/object/src/api.js

```diff
@@ -10,4 +10,27 @@ let builder = new APIBuilder({
   context: ['cfg', 'db', 'backends'],
 });
 
+builder.declare({
+  method: 'post',
+  route: '/upload/:name',
+  name: 'uploadObject',
+  input: 'upload-object-request.yml',
+  stability: 'experimental',
+  category: 'Upload',
+  scopes: 'object:upload:<name>',
```

Comment:
I think we will want to be able to limit uploads by projectId, so this should probably be
```suggestion
  scopes: 'object:upload:<projectId>:<name>',
```

Reply:
Removing `projectId` from the route but adding it here throws:

```
  services/object/test/api_test.js
    ✓ ping
    1) should be able to upload
    2) "after each" hook for "should be able to upload"


  1 passing (2s)
  2 failing

  1) services/object/test/api_test.js
       should be able to upload:
     Internal Server Error, incidentId 38f16edb-f25e-4ce7-be7d-6e0a1c95be9d. Error (not shown in production):
```
Error: Deferred auth was never checked!
    at ServerResponse.res.reply (/Users/hassanali/Documents/Mozilla/projects/taskcluster/libraries/api/src/middleware/schema.js:72:15)
    at Proxy.<anonymous> (/Users/hassanali/Documents/Mozilla/projects/taskcluster/services/object/src/api.js:33:14)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

Reply:
Yeah, we'd need to call `req.authenticate` in that case since projectId is in the request body.